### ISSUE TITLE
3.0 Bug List Checklist

### ISSUE THREAD

**Issue by Nicholas-Cardot** [warfare-plugins/social-warfare#532]
- [x] Test all the share buttons 
- [x] Remove all the legacy files.
- [x] Add generate_share_link() method to the subclasses.
- [x] Figure out why shortcodes aren't working on the home page.
- [x] Make a migration script for the post editor options.
- [x] Map the side floating buttons to the new class.
- [x] Get rid of the rogue ` in the settings page sidebar
- [x] Buttons style preview is not working on settings page
- [x] Color picker for custom colors doesn't show on Style settings page
- [x] Ensure the floating buttons are working.
- [x] Ensure the buttons are APPENDED to the content and not replacing the content.
 - [x] Make sure the totals floats to the right when set to the right.
  - [x] Normalize all the options back to how they were.
 - [x] Add the "other" button hover option. It seems to be missing.
 - [x] Ensure that the social_warfare() function acts as a passthrough instantiating and rendering the class based version and passing along any arguments that were passed into it.
 - [x] Ensure shortcodes are generating buttons properly.
 - [x] Make the panel use the share count function.
- [x] Get the shortcodes working.
- [x] Location options for home page and archives are missing from the options page.
- [x] Remove the "boxed" option on the static buttons.
- [x] Fix the LinkedIn icon in the CSS.

Note: If you file a bug report below, a thumbs up will indicate that I have now fixed the issue reported in that comment.

**Comment by DustinWStout** [comment]
The "boxed" style is showing as an available option for the static buttons but when selected the style remains as "Flat and Fresh".

**Comment by nutsandbolts** [comment]
I lost all existing settings and options when I installed 3.0.0 - definitely not something we want to happen to customers.

**Comment by nutsandbolts** [comment]
Errors on activation (PHP 7.2)

`[Wed May 02 18:09:59.006416 2018] [proxy_fcgi:error] [pid 6986] [client 127.0.0.1:57001] AH01071: Got error 'PHP message: PHP Notice:  Undefined index: float_location in /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-content/plugins/social-warfare/functions/frontend-output/SWP_Display.php on line 34\nPHP message: PHP Stack trace:\nPHP message: PHP   1. {main}() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-admin/admin-ajax.php:0\nPHP message: PHP   2. require_once() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-admin/admin-ajax.php:22\nPHP message: PHP   3. require_once() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-load.php:37\nPHP message: PHP   4. require_once() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-config.php:74\nPHP message: PHP   5. include_once() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-settings.php:305\nPHP message: PHP   6. Social_Warfare->__construct() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-content/plugins/social-warfare/social-warfare.php:29\nPHP message: PHP   7. Social_Warfare->instantiate_classes() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-content/plugins/social-warfare/functions/Social_Warfare.php:34\nPHP message: PHP   8. SWP_Display->__construct() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-content/plugins/social-warfare/functions/Social_Warfare.php:125\nPHP message: PHP Warning:  A non-numeric value encountered in /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-content/plugins/social-warfare-pro/functions/admin/SWP_Pro_Options_Page.php on line 326\nPHP message: PHP Stack trace:\nPHP message: PHP   1. {main}() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-admin/admin-ajax.php:0\nPHP message: PHP   2. require_once() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-admin/admin-ajax.php:22\nPHP message: PHP   3. require_once() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-load.php:37\nPHP message: PHP   4. require_once() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-config.php:74\nPHP message: PHP   5. do_action() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-settings.php:471\nPHP message: PHP   6. WP_Hook->do_action() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-includes/plugin.php:453\nPHP message: PHP   7. WP_Hook->apply_filters() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-includes/class-wp-hook.php:310\nPHP message: PHP   8. Social_Warfare_Pro->instantiate_addon() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-includes/class-wp-hook.php:286\nPHP message: PHP   9. SWP_Pro_Options_Page->__construct() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-content/plugins/social-warfare-pro/functions/Social_Warfare_Pro.php:59\nPHP message: PHP  10. SWP_Pro_Options_Page->update_social_tab() /home/138698.cloudwaysapps.com/zfythsrwzs/public_html/wp-content/plugins/social-warfare-pro/functions/admin/SWP_Pro_Options_Page.php:15\n', referer: https://betweenthebolts.com/wp-admin/plugins.php
`

**Comment by DustinWStout** [comment]
*Clean Install of WordPress*

Activated both Social Warfare and Pro and when I registered my license key and hit "Save Changes" the only Pro features that showed up at first were the additional buttons. I had to refresh the page in order for the other Pro features to show.

**Comment by Nicholas-Cardot** [comment]
Cool. That's great to know. I can fix that easily. I think I already know exactly what needs to be done.

**Comment by nutsandbolts** [comment]
Also, when I opened Inspector to look at the above error, I lost the button preview completely. The whole section was blank until I refreshed the page.

**Comment by Nicholas-Cardot** [comment]
11. I don't think we've ever had a proper fallback for when you click "deny" on bitly so that is nothing new. We can add that to the wishlist.

15. I'm not sure how to detect if they have it installed or not. But this isn't new so it's something we can consider for future research.

**Comment by nutsandbolts** [comment]
We should probably make Pro dependent on core, so if someone tries to install Pro without core, they get a message telling them core is required. I do this all the time when I require plugins to be installed with a custom theme: [https://wisdmlabs.com/blog/wordpress-deactivate-plugin-depending-on-another-activation/](https://wisdmlabs.com/blog/wordpress-deactivate-plugin-depending-on-another-activation/)

**Comment by Nicholas-Cardot** [comment]
We'll work on a message later. Right now, the way it works is that if core is not loaded or if it's not on a current version of core, then pro simply doesn't load any of it's files or functions. But a message is a good idea.

**Comment by DustinWStout** [comment]
@ckmahoney re: Three-dee styling in Styles preview...

![three-dee-styling-preview](https://user-images.githubusercontent.com/2152604/39726360-a2d45e40-5203-11e8-8c75-f09be59fe838.jpg)

So I unchecked that box in my previous comment.

**Comment by Nicholas-Cardot** [comment]
This is because the line height is blocking the overflow for some reason. I'm not sure how exactly to fix it that won't break the front end display.

**Comment by Nicholas-Cardot** [comment]
The height of the panel is 2px shorter than the space needed to show the extra border bottom.

**Comment by DustinWStout** [comment]
From one of our Beta Testers (Lisa): 

- [ ] When I’ve tested using the AffiliateWP integration, it does not work with Facebook, there is no affiliate info.

So in other words, we need to test all the buttons AGAIN with Social Warfare - AffiliateWP activated.
